raiden_contracts/deploy/etherscan_verify.py

```diff
@@ -90,24 +92,23 @@ def etherscan_verify(chain_id, apikey, guid, contract_name):
         )
 
     if contract_name is None or contract_name == CONTRACT_TOKEN_NETWORK_REGISTRY:
-        source = (
-            source_path['raiden'].joinpath('Utils.sol').read_text() +
-            source_path['raiden'].joinpath('Token.sol').read_text() +
-            source_path['raiden'].joinpath('lib', 'ECVerify.sol').read_text() +
-            source_path['raiden'].joinpath('SecretRegistry.sol').read_text() +
-            source_path['raiden'].joinpath('TokenNetwork.sol').read_text() +
-            source_path['raiden'].joinpath('TokenNetworkRegistry.sol').read_text()
-        )
-
-        source = 'pragma solidity ^0.4.23;' + source.replace('pragma solidity ^0.4.23;', '')
-        source = source.replace('import "raiden/Token.sol";\n', '')
-        source = source.replace('import "raiden/Utils.sol";\n', '')
-        source = source.replace('import "raiden/lib/ECVerify.sol";\n', '')
-        source = source.replace('import "raiden/SecretRegistry.sol";\n', '')
-        source = source.replace('import "raiden/TokenNetwork.sol";\n', '')
+        joined_file = Path(__file__).parent.joinpath('joined.sol')
+        command = [
+            './utils/join-contracts.py',
+            '--import-map',
+            '{"raiden": "contracts", "test": "contracts/test", "services": "contracts/services"}',
+            'contracts/TokenNetworkRegistry.sol',
+            str(joined_file),
+        ]
+        old_working_dir = Path.cwd()
+        chdir(Path(__file__).parent.parent)
+        try:
+            combiner = subprocess.Popen(command)
+            combiner.wait()
+        finally:
+            chdir(old_working_dir)
```

Comment:
You can avoid the `wait` call by using `subprocess.run` and the directory handling by using the `pwd` parameter:
```python
    subprocess.run(command, cwd=Path(__file__).parent.parent)
```

Reply:
I already touch this part in the next PR.

I didn't know `cwd` keyword.  It is really neat.  I'll use it there.